spec/subjects/async-subject-spec.js

```diff
@@ -52,34 +55,50 @@ describe('AsyncSubject', function () {
     expect(results).toEqual([2, 'done']);
   });
 
-  it('should just complete if no value has been nexted into it', function () {
+  it('should not emit values after complete', function () {
+    var subject = new AsyncSubject();
+
+    var subscription = subject.subscribe(observer);
+
+    subject.next(1);
+    expect(results).toEqual([]);
+    subject.next(2);
+    expect(results).toEqual([]);
+    subject.complete();
+    expect(results).toEqual([2, 'done']);
+    subject.next(3);
+    expect(results).toEqual([2, 'done']);
+  });
+
+  it('should not emit values if unsubscribed before complete', function () {
```

Comment:
first case seems right, I missed :)

actually for second, I think I wrote snippet incorrectly.

```
subject.subscribe(subscriber);
subscriber.unsubscribe(); -> subscriber has unsubscribed explicitly
subject.next('boom');
```

is this already covered also?